week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/product/ProductFree.ts

```diff
@@ -0,0 +1,11 @@
+import { Product, PRODUCT_TYPE } from './Product';
+
+class ProductFree extends Product {
+  #type: PRODUCT_TYPE = PRODUCT_TYPE.free;
```

Comment:
Jak dla mnie @DariuszGala to to `type` przestaje być w tym momencie potrzebne.

Zauważ że teraz klasa świadczy sama o sobie jakim jest typem.
Chodzi o to że operatorem `instanceof` zobaczysz co to jest

np. 

```typescript
const product = new ProductFree(/*....*/)

console.log(product instanceof Product) // true

console.log(product instanceof ProductFree) // true

console.log(product instanceof ProductWithPrice) // false
```